clang/test/CodeGen/X86/fp128-abi.c

```diff
@@ -0,0 +1,35 @@
+// RUN: %clang_cc1 -triple x86_64-linux -emit-llvm  -target-feature +sse2 < %s | FileCheck %s --check-prefixes=CHECK
```

Comment:
Worth adding a non-SSE RUN?

Reply:
This patch only changes for 64-bit ABI, non-SSE is not a valid case for 64-bit.
OTOH, -sse just generate identical output with +sse2, which doesn't match with GCC https://godbolt.org/z/4nxnhnovd
I think we may need to add a semacheck for it. Do you think we should add it with this patch or a follow up?

Reply:
If you're dealing with it soon then handling it in a followup sounds good to me

Reply:
I decided not to report error for `-sse` after investigating the current diagnosis machinism.
We didn't report the SSE error in semacheck but in backend due to backend crash.
`__float128` doesn't have crash issue and can be passed on GPR registers without SSE enabled.
IIUC, we prefer to be compatible with early version rather than always to GCC. So I'd like to keep the convention as is.

Reply:
OK - so we have test coverage for non-SSE cases already in the backend?

Reply:
Yes, e.g, `llvm/test/CodeGen/X86/{soft-fp,x87}.ll`, though I doubt if they can work in reality since they are calling to the same lib functions.